harness/determined/common/api/authentication.py

```diff
@@ -62,10 +62,11 @@ def _init_session(
         session_user = (
             requested_user
             or self.token_store.get_active_user()
+            or util.get_initial_user()
             or constants.DEFAULT_DETERMINED_USER
         )
 
-        token = self.token_store.get_token(session_user)
+        token = self.token_store.get_token(session_user) or util.get_user_token()
         if token is not None and not _is_token_valid(self.master_address, token, cert):
             self.token_store.drop_user(session_user)
             token = None
```

Comment:
I think this line:

```python
if token is not None and not _is_token_valid(self.master_address, token, cert):
```

should not apply to the session container token.  Maybe:

```python
        token = self.token_store.get_token(session_user)
        if token is not None and not _is_token_valid(self.master_address, token, cert):
            self.token_store.drop_user(session_user)
            token = None
        # do it here instead:
        if token is None:
            token = util.get_user_token()
```